apm-lambda-extension/extension/process_env.go

```diff
@@ -62,6 +66,39 @@ func getIntFromEnv(name string) (int, error) {
 	return value, nil
 }
 
+func getSecret(secretName string) (string, error) {
+	region := os.Getenv("AWS_REGION")
+
+	sess, err := session.NewSession()
+	if err != nil {
+		return "", err
+	}
+	svc := secretsmanager.New(sess, aws.NewConfig().WithRegion(region))
+	input := &secretsmanager.GetSecretValueInput{
+		SecretId:     aws.String(secretName),
+		VersionStage: aws.String("AWSCURRENT"),
+	}
+
+	result, err := svc.GetSecretValue(input)
+	if err != nil {
+		return "", err
+	}
+
+	var secretString string
+	if result.SecretString != nil {
+		secretString = *result.SecretString
+	} else {
+		decodedBinarySecretBytes := make([]byte, base64.StdEncoding.DecodedLen(len(result.SecretBinary)))
+		len, err := base64.StdEncoding.Decode(decodedBinarySecretBytes, result.SecretBinary)
```

Comment:
`len` is a builtin and, while the compiler might let you use it, it'd probably be better to choose a different variable name